### PR TITLE
Adding NO_PROFILE checks before creating MercProfile object

### LINKED ISSUE #1332: Quest Debug System crashes most of the time
[//]: # (Thank you for taking the time to report an issue or propose an enhancement. Questions are also welcome.)
[//]: # (If you have more than one item to report, please open separate issues!)

[//]: # (This text will not be visible and you can freely delete it. Check the Preview tab for a cleaner look.)

### What is wrong
If you try to start the Quest Debug System and there is anyone or anything present in the sector without a merc profile the QDS crashes with the message "_Game has been terminated due to an unrecoverable error: invalid m_profileID '200'_". As you probably know, NO_PROFILE is defined as 200 in `Soldier_Control.h`

Since the QDS is actually useful when it's working and is mentioned on the [features page](https://ja2-stracciatella.github.io/features/) it'd be nice to get at least the basic functionality working again. It's also quite crashy when you let characters speak, but that has been the case since vanilla.

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  v0.17.1-git+866226846 | JA2 Gold Eng.  |    no    |    yes    |

### PR DESCRIPTION
**Pull request by ja2-derek**
Creating MercProfile object with invalid ID will result in an exception. This has caused problems in the QDS (and maybe some other places).

Resolves #1332.

### CODE REVIEW

**Comment by ja2-derek** [comment]
This fixes the crash on opening the QDS.

File: src/game/Strategic/Strategic_Merc_Handler.cc [hunk 1]
```diff
@@ -177,6 +177,7 @@ void MercDailyUpdate()
 			if (--s->bCorpseQuoteTolerance < 0) s->bCorpseQuoteTolerance = 0;
 
 			MERCPROFILESTRUCT& p = GetProfile(s->ubProfile);
+			MercProfile const profile(s->ubProfile);
```
Comment (lynxlynxlynx) [inline src/game/Strategic/Strategic_Merc_Handler.cc:180]: Does this internalize the check or why isn't it needed here?

Comment (ja2-derek) [reply src/game/Strategic/Strategic_Merc_Handler.cc:180]: [GetProfile](https://github.com/ja2-stracciatella/ja2-stracciatella/blob/bee3fe37530d618b7c391bc7e85cf955369a8626/src/game/Tactical/Soldier_Profile.h#L14-L20) checks for a valid profile ID, and aborts on invalid ID.

Comment (lynxlynxlynx) [reply src/game/Strategic/Strategic_Merc_Handler.cc:180]: Still quite strict, but it's unlikely to be hit in this place.

**Comment by lynxlynxlynx** [comment]
Thanks!

**Pull request merged**